catalog/transformer/catalog-transformer-service-atom/src/main/java/ddf/catalog/transformer/response/query/atom/AtomTransformer.java

```diff
@@ -233,156 +223,177 @@ public BinaryContent transform(SourceResponse sourceResponse, Map<String, Serial
       startIndex.setText(Integer.toString(sourceResponse.getRequest().getQuery().getStartIndex()));
     }
 
-    for (Result result : sourceResponse.getResults()) {
-
-      Metacard metacard = result.getMetacard();
-
-      if (metacard == null) {
-        continue;
-      }
-
-      Entry entry = feed.addEntry();
-
-      String sourceName = DEFAULT_SOURCE_ID;
-
-      if (result.getMetacard().getSourceId() != null) {
-        sourceName = result.getMetacard().getSourceId();
-      }
+    if (getCount(sourceResponse) != 0) {
+      sourceResponse.getResults().stream().forEach(r -> addSingleResult(currentDate, feed, r));
+    }
 
-      Element source =
-          entry.addExtension(new QName(FEDERATION_EXTENSION_NAMESPACE, "resultSource", "fs"));
+    byte[] bytes = createOutputStream(feed);
+    return new BinaryContentImpl(new ByteArrayInputStream(bytes), MIME_TYPE);
+  }
 
-      /*
-       * According to the os-federation.xsd, the resultSource element text has a max length of
-       * 16 and is the shortname of the source id. Previously, we were duplicating the names
-       * in both positions, but since we truly do not have a shortname for our source ids, I
-       * am purposely omitting the shortname text and leaving it as the empty string. The real
-       * source id can still be found in the attribute instead.
-       */
+  private byte[] createOutputStream(Feed feed) throws CatalogTransformerException {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
❓ Shouldn't this be a `FileBackedOutputStream` to help guard against memory usage issues?

Reply:
This was a refactoring, old version had a ByteArrayOutputStream, this was not a change.  I think our query restrictions keep this from being a risk with memory issues.  Switching to a filebacked stream would be ok for safety, but I do not expect it would ever actually flip over to use a file.

Reply:
I am thinking of users with extremely large metacards, but I agree that it is unlikely.

Reply:
Given that this is not a new change, are we comfortable with it as-is?

Reply:
Yup.